File: packages/silo-client/src/media/media-stream.ts
import { RowStream } from "../pagination/row-stream.js";
import type { MediaAsset } from "./media-asset.js";

/** `Media.all()`'s async-iterable: one `MediaAsset` at a time, paging by the
 * window the server echoes. No behaviour of its own — the named type
 * is what `Media.all()` answers. */
export class MediaStream extends RowStream<MediaAsset> {}

File: packages/silo-client-java/src/main/java/in/org/quicko/silo/client/media/MediaStream.java
package in.org.quicko.silo.client.media;

import in.org.quicko.silo.client.RequestOptions;
import in.org.quicko.silo.client.pagination.RowLoader;
import in.org.quicko.silo.client.pagination.RowStream;

/**
 * What {@code media.all()} answers: one asset at a time, across every page. No
 * behaviour of its own — the named type is what makes the return value readable.
 */
public final class MediaStream extends RowStream<MediaAsset> {
  public MediaStream(RowLoader<MediaAsset> loader, int startingLimit, RequestOptions options) {
    super(loader, startingLimit, options);
  }
}
